### ISSUE TITLE
Cookbook Bulk upload option on supermarket

### ISSUE THREAD

**Issue by ishug86** [chef/knife-supermarket#26]
We have installed Private Supermarket in our organisation, What is the command that I can use to upload multiple cookbooks at once? 
We have around 300 cookbooks in that we want to upload on our private supermarket instance.

**Comment by nellshamrell** [comment]
Unfortunately, there is not currently a way to bulk upload to Supermarket beyond using a shell script to run the knife supermarket upload command on each cookbook in a directory.

**Comment by ishug86** [comment]
Thanks for the reply , also I am not able to upload the cookbook to supermarket for a specific version , however I am able to unshare it for a version. Is this functionality not present for share?

**Comment by nellshamrell** [comment]
What tool are you using to upload cookbooks?  Knife Supermarket?  Stove?

**Comment by ishug86** [comment]
knife share , since I am deploying on private supermarket .

985aeb8b6b92:supermarket_second zng937$ knife supermarket share "007_JavaMSP_Apache_Admintool_Installation/versions/1.0.4" -c knife.rb 
ERROR: Chef::Exceptions::CookbookNotFoundInRepo: Cannot find a cookbook named 007_JavaMSP_Apache_Admintool_Installation/versions/1.0.4; did you forget to add metadata to a cookbook? (https://docs.chef.io/config_rb_metadata.html)

**Comment by nellshamrell** [comment]
Does it work when you run:

knife supermarket share "007_JavaMSP_Apache_Admintool_Installation" -c knife.rb 

(I know this will not share the version you wish to, but let's try this just for troubleshooting)

**Comment by ishug86** [comment]
Yes , it runs , and it picks the latest version.

**Comment by nellshamrell** [comment]
Got it.  There is unfortunately not currently a way to share a specific version of a cookbook on Supermarket, it will pick up the latest version.  If you'd like, we can look at adding this as a feature, but it will need to be prioritized with our current workload.

**Comment by ishug86** [comment]
Okay , thanks for the Info. I think bulk upload and upload based on the latest version will surely help other developers to sync their cookbooks to private supermarket.

**Comment by lamont-granquist** [comment]
needs to be opened against supermarket, not knife-supermarket

**Comment by JackChance** [comment]
I know this was closed but I had a similar issue and this was one of the top results on Google. The fix I found uses bash to recursively apply the supermarket share command to each individual cookbook in a directory

apply this command in your cookbook repo directory. It will fail on any folder that isn't a cookbook that you might have (such as a readme or a .rubocop.yml), and there is plenty of room to optimize it.
`for file in *; do knife supermarket share "$file"; done`
